app/controllers/api/api_controller.rb

```diff
@@ -101,6 +97,120 @@ def dashboard
     end
   end
 
+  # Get Team & Team Role & Permissions info for the Mobile App.
+  def mobile_basic
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    user_team = Team.find_by_id(current_user.team_id)
+    user_team_role = TeamRole.find_by_id(current_user.team_role_id)
+
+    begin
+      #generate overall embedded result
+      @result = {
+        'team' => user_team,
+        'team_role' => user_team_role
+      }
+    rescue
+      @status = 500
+      @message = "Failure to generate mobile basic results."
+    end
+
+
+    # if we haven't set a message, we sucessfully made stuff
+    unless @message
+      @status = 200
+      @message = "Success!"
+    end
+
+    @response = {
+      status: @status,
+      message: @message,
+      date:  Time.now,
+      result: @result
+    }
+
+    respond_to do |format|
+      format.json { render :json => @response }
+    end
+  end
+
+  # Get Public Relations / Income info for the Mobile App - Income Tab.
+  def income
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    round = @game.round
+    public_relations_list = PublicRelation.where(game: @game)
+
+    begin
+      #generate overall embedded result
+      @result = {
+        "timer" => {
+          "round"=>  round,
+        },
+        "pr" => public_relations_list
+      }
+    rescue
+      @status = 500
+      @message = "Failure to generate public relation results."
+    end
+
+    # if we haven't set a message, we sucessfully made stuff
+    unless @message
+      @status = 200
+      @message = "Success!"
+    end
+
+    @response = {
+      status: @status,
+      message: @message,
+      date:  Time.now,
+      result: @result
+    }
+
+    respond_to do |format|
+      format.json { render :json => @response }
+    end
+  end
+
+ # Get Message Data for the Mobile App - Messages Tab.
+  def messages
+    @game = params[:game_id].nil? ? current_game : Game.find_by_id(params[:game_id])
+    @game.update
+
+    round = @game.round
+    messages = Message.all.order('created_at DESC')
```

Comment:
So a conversation object is just made up of all the messages between two nations as determined by sender/recipient id.

Well, this gives you all of the messages...
`messages = Message.where(game: @game).where("recipient_id = ? or sender_id = ?", user_team_id, user_team_id)`

The problem is grouping based on sender_id and also recipient_id.  I'm not sure if that's possible to do in one query.  Or even at all since it's group_by with an or conditional...

What comes to mind is to do a distinct on recipient_id and sender_id on the above query and then merge the results, excluding user_team_id.  Then for each participant_id, select also on whether the recipient_id or sender_id is equal to the participant_id.  Technically this is n+1 queries but as we will never have an absurd number of recipient/sender id's (meaning number of teams), I think it's okay.

Come to think of it, you could more quickly get distinct participant_id's by checking the user table filtering on game_id and getting distinct team id's?  Although that raises another question for me which is how you can use sender_id and recipient_id in messages when they are both strings in the schema.  Suggests you have a migration there that I'm not aware of...  If we correctly tie sender_id and recipient_id to a team_id... my instincts say there is a potential join solution but it's probably too complex to spend time on/out of scope.